### PR TITLE
[backend/elfobj] refactor symbol index emission in obj_rtinit

### PR DESCRIPTION
**Pull request by thewilsonator**
in support of #12429

### CODE REVIEW

File: src/dmd/backend/elfobj.d [hunk 1]
```diff
@@ -3392,10 +3392,7 @@ private void obj_rtinit()
     // make the symbols hidden so that each DSO gets its own brackets
     IDXSYM minfo_beg, minfo_end, dso_rec;
 
-    static if (TARGET_OPENBSD)
-    {
-        IDXSYM deh_beg, deh_end;
-    }
+    IDXSYM deh_beg, deh_end;
```
Comment (RazvanN7) [inline src/dmd/backend/elfobj.d:3395]: Why is the `static if` dropped?

Comment (thewilsonator) [reply src/dmd/backend/elfobj.d:3395]: #12429

**Comment by RazvanN7** [comment]
cc @WalterBright

**Review by WalterBright — DISMISSED** [review]
I have gone through the backend and removed nearly all use of TARGET_XXXXX constants in the service of making it possible to use dmd as a cross compiler.

So, please do not reintroduce such.

**Comment by WalterBright** [comment]
Use `config.exe` instead, which is a runtime variable.

**Comment by thewilsonator** [comment]
@WalterBright done

**Pull request merged**